src/cam_gst.cpp

```diff
@@ -253,7 +258,12 @@ bool CamGst::getBuffer(std::vector<uint8_t>& buffer, bool blocking_read,
         } else {
             // Copy buffer for return.
             buffer.resize(mBufferSize);
-            memcpy(&buffer[0], GST_BUFFER_DATA(mBuffer), mBufferSize);
+            GstMapInfo info;
+            GstMapFlags flags = GST_MAP_READ;
+            GstBuffer* b = gst_sample_get_buffer(mSample);
+            gst_buffer_map(b, &info, flags);
```

Comment:
return value of gst_buffer_map (gbool) should be checked

Reply:
return value is now checked